addons/openvpn/openvpn-client-dep.yaml

```diff
@@ -119,6 +119,11 @@ spec:
             memory: 50Mi
       restartPolicy: Always
       terminationGracePeriodSeconds: 5
+      tolerations:
+       - effect: NoSchedule
+         operator: Exists
+       - effect: NoExecute
+         operator: Exists
```

Comment:
These "blanket" tolerations (for OpenVPN and all others in this PR) can be problematic because cordoning off a node doesn't work properly anymore (`kubectl cordon` and `kubectl drain` set the taint `node.kubernetes.io/unschedulable:NoSchedule`). This is fine for `DaemonSets` (envoy-agent, node-local-dns, promtail), but it can cause Pods to go right back onto the node after being evicted (as no taint can prevent them from being scheduled), causing an endless loop that never fully drains a node.

Reply:
right we were going back and forth with whether to add that to deployment

Reply:
but what about case when user add only one machine-deployment with some taints, those system components won't be ever scheduled

Reply:
@aborilov that's correct, but only creating one MachineDeployment and then adding taints to it doesn't make any sense -  The idea of taints is to exclude Pods from a set of nodes, so _they run elsewhere_. If there are no nodes without taints, all Pods need a toleration, and taints and tolerations offset each other.

What we could do is add a special toleration to system components (`kkp.k8c.io/system:NoSchedule` or something) and document that in the KKP documentation as necessary if you want to create your initial MachineDeployment as a "system-only" pool of nodes, you need to use that toleration. Alternatively, we could document that the initial MachineDeployment must not have custom taints.

The problem is essentially that we can't do "ignore all taints, except the ones coming from k8s".

Reply:
so, let's leave tolerations only for daemon sets, that should work, right?

Reply:
Yeah, that should do the trick!